### PR TITLE
allow totalcount to be string.

### PR DESCRIPTION
**Pull request by cyjia**
With totalCount be String, `new Array("100")` will result in an Array with first element `"100"`.

### CODE REVIEW

File: addon/models/lazy-array.js [hunk 1]
```diff
@@ -60,6 +59,10 @@ export default Ember.ArrayProxy.extend({
     }
   },
 
+  _totalCount: Ember.computed(function() {
+    return parseInt(this.get('totalCount'));
+  }).property('totalCount'),
+
```
Comment (jefflage) [inline addon/models/lazy-array.js]: why do we need to protect for `totalCount` being a string?

Comment (cyjia) [inline addon/models/lazy-array.js]: When initializing a lazy array, a empty array with length of `totalCount` will be created. If we use `new Array(this.get('totalCount'))` and the totalCount passed in is a string, then the result array's first element is the value of `this.get('totalCount')`. We spent some time to find the issue, so we want to encapsulate this knowledge and prevent others from being trapped in this again.

Comment (cyjia) [inline addon/models/lazy-array.js]: We grabbed a query parameter from url and passed to LazyArray as totalCount, then we found first row of table is always blank. We wrote a test to verify that but tests passed. Only after hours of  checking code and hacking code, then we realized the totalCount from url is a string. Maybe someone else will make the same mistake while using ember-table.

Comment (jefflage) [inline addon/models/lazy-array.js]: Brilliant.  Thanks for the good explanation

**Pull request merged**